src/lib/hooks/useZoom.ts

```diff
@@ -17,6 +17,16 @@ export const useZoom = ({
 }): { scale: number; resetScale: () => void; setNewScale: (newScale: number) => void } => {
     const [scale, setScale] = React.useState(1);
 
+    const interval = React.useRef<null | ReturnType<typeof setInterval>>();
+
+    React.useEffect(() => {
+        return () => {
+            if (interval.current) {
+                clearInterval(interval.current);
+            }
+        };
+    }, []);
+
     React.useEffect(() => {
         const handleWheelEvent = (e: WheelEvent) => {
             e.preventDefault();
```

Comment:
Consider adding clearing of interval in `handleWheelEvent`:
```
if (interval.current) {
    clearInterval(interval.current);
}
```

A double click on map will trigger `doubleClickEvent` zoom in with interval counter. If the user performes zoomOut with wheel event before `interval.current` is cleared - the `doubleClickEvent` will then zoom in after a zoom out scroll is performed.